### ISSUE TITLE
"TemplateSyntaxError: 'future' is not a registered tag library" for SortableTabularInline

### ISSUE THREAD

**Issue by EugeneDae** [iambrandontaylor/django-admin-sortable#151]
I'm using Python 3.5.2, Django 1.10 and django-admin-sortable 2.0.18. I've also tried master version from this repository.

```
Exception Value: 'future' is not a registered tag library. Must be one of:
admin_list
admin_modify
admin_static
admin_urls
adminsortable_tags
cache
django_template_additions
i18n
l10n
log
static
staticfiles
thumbnail
tz
```

```
 Traceback:

File "[...]/lib/python3.5/site-packages/django/template/defaulttags.py" in find_library
  1021.         return parser.libraries[name]

During handling of the above exception ('future'), another exception occurred:

File "[...]/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  217.                 response = self.process_exception_by_middleware(e, request)

File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  215.                 response = response.render()

File "[...]/lib/python3.5/site-packages/django/template/response.py" in render
  109.             self.content = self.rendered_content

File "[...]/lib/python3.5/site-packages/django/template/response.py" in rendered_content
  86.         content = template.render(context, self._request)

File "[...]/lib/python3.5/site-packages/django/template/backends/django.py" in render
  66.             return self.template.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  208.                     return self._render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  174.         return compiled_parent._render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  174.         return compiled_parent._render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  174.         return compiled_parent._render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in _render
  199.         return self.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  70.                 result = block.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  70.                 result = block.nodelist.render(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render
  994.                 bit = node.render_annotated(context)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/defaulttags.py" in render
  217.                     nodelist.append(node.render_annotated(context))

File "[...]/lib/python3.5/site-packages/django/template/base.py" in render_annotated
  961.             return self.render(context)

File "[...]/lib/python3.5/site-packages/django/template/loader_tags.py" in render
  201.                     template = context.template.engine.get_template(template_name)

File "[...]/lib/python3.5/site-packages/django/template/engine.py" in get_template
  160.         template, origin = self.find_template(template_name)

File "[...]/lib/python3.5/site-packages/django/template/engine.py" in find_template
  134.                         name, template_dirs=dirs, skip=skip,

File "[...]/lib/python3.5/site-packages/django/template/loaders/base.py" in get_template
  44.                     contents, origin, origin.template_name, self.engine,

File "[...]/lib/python3.5/site-packages/django/template/base.py" in __init__
  191.         self.nodelist = self.compile_nodelist()

File "[...]/lib/python3.5/site-packages/django/template/base.py" in compile_nodelist
  233.             return parser.parse()

File "[...]/lib/python3.5/site-packages/django/template/base.py" in parse
  518.                     raise self.error(token, e)

File "[...]/lib/python3.5/site-packages/django/template/base.py" in parse
  516.                     compiled_result = compile_func(self, token)

File "[...]/lib/python3.5/site-packages/django/template/defaulttags.py" in load
  1072.         lib = find_library(parser, name)

File "[...]/lib/python3.5/site-packages/django/template/defaulttags.py" in find_library
  1025.                 name, "\n".join(sorted(parser.libraries.keys())),
```

Also SortableStackedInline works ok.

**Comment by EugeneDae** [comment]
A simple workaround for Django 1.10 based projects is to create a stub for `future` tag library:

```
# templatetags/future.py
from django.template import Library
from django.template.defaulttags import cycle as cycle_original

register = Library()


@register.tag
def cycle(*args, **kwargs):
    ''' A stub to get SortableTabularInline to work '''
    return cycle_original(*args, **kwargs)

```

**Comment by alsoicode** [comment]
Hello. Sorry I haven't had a chance to look at this yet, but I will try to do so this evening.

**Comment by EugeneDae** [comment]
With my workaround, I still can't sort SortableTabularInline - the ajax request results in this:

```
Internal Server Error: /cms/locations/place/7/change/
Traceback (most recent call last):
  File "[...]/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 544, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/sites.py", line 211, in inner
    return view(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/adminsortable/admin.py", line 255, in change_view
    form_url='', extra_context=extra_context)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1512, in change_view
    return self.changeform_view(request, object_id, form_url, extra_context)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/usr/local/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1448, in changeform_view
    if all_valid(formsets) and form_validated:
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 456, in all_valid
    if not formset.is_valid():
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 321, in is_valid
    self.errors
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 295, in errors
    self.full_clean()
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 342, in full_clean
    for i in range(0, self.total_form_count()):
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 116, in total_form_count
    return min(self.management_form.cleaned_data[TOTAL_FORM_COUNT], self.absolute_max)
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 98, in management_form
    code='missing_management_form',
django.core.exceptions.ValidationError: ['ManagementForm data is missing or has been tampered with']
17:07:26 ERROR Internal Server Error: /cms/locations/place/7/change/
Traceback (most recent call last):
  File "[...]/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "[...]/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 544, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/sites.py", line 211, in inner
    return view(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/adminsortable/admin.py", line 255, in change_view
    form_url='', extra_context=extra_context)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1512, in change_view
    return self.changeform_view(request, object_id, form_url, extra_context)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "[...]/lib/python3.5/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/usr/local/Cellar/python3/3.5.2_1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/contextlib.py", line 30, in inner
    return func(*args, **kwds)
  File "[...]/lib/python3.5/site-packages/django/contrib/admin/options.py", line 1448, in changeform_view
    if all_valid(formsets) and form_validated:
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 456, in all_valid
    if not formset.is_valid():
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 321, in is_valid
    self.errors
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 295, in errors
    self.full_clean()
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 342, in full_clean
    for i in range(0, self.total_form_count()):
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 116, in total_form_count
    return min(self.management_form.cleaned_data[TOTAL_FORM_COUNT], self.absolute_max)
  File "[...]/lib/python3.5/site-packages/django/forms/formsets.py", line 98, in management_form
    code='missing_management_form',
django.core.exceptions.ValidationError: ['ManagementForm data is missing or has been tampered with']

```

Interestingly, when I simply replace `SortableTabularInline` with `SortableStackedInline`, everything works perfectly.

Thank you for your work, Brandon.

**Comment by vstoykov** [comment]
If usage of [`{% load cycle from future %}` in `adminsortable/edit_inline/tabular.html`](https://github.com/iambrandontaylor/django-admin-sortable/blob/master/adminsortable/templates/adminsortable/edit_inline/tabular.html#L1) is somehow avoided then it will work in Django 1.10 too.

**Comment by alsoicode** [comment]
Sorry for the delay getting to this bug, I've been slammed. I should have
some time this evening to look at it.

**Comment by alsoicode** [comment]
@EugeneDae A quick fix to get you going is to simply delete the `{% load cycle from future %}` template tag. This is just some left over legacy code. I see that the inline form templates have changed a bit in 1.10, so I'll need to extend those and make sure everything works before I can create a permanent fix.

**Comment by alsoicode** [comment]
Ok, I have released version 2.0.19 to PyPi that fixes this issue and updates the admin templates to be more compatible with 1.10
